shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala

```diff
@@ -688,6 +700,10 @@ object GlutenConfig {
       (SPARK_S3_USE_INSTANCE_CREDENTIALS, "false"),
       (SPARK_S3_IAM, ""),
       (SPARK_S3_IAM_SESSION_NAME, ""),
+      (SPARK_S3_RETRY_MAX_ATTEMPTS, "3"),
+      (SPARK_S3_CONNECTION_MAXIMUM, "96"),
```

Comment:
`fs.s3a.attempts.maximum` and `fs.s3a.connection.maximum` can also be aligned with hadoop config, default value is 20 and 15.